client/src/components/portfolio/AddTransaction.js

```diff
@@ -0,0 +1,119 @@
+import React, { Component } from "react";
+import { Link, withRouter } from "react-router-dom";
+import TextFieldGroup from "../common/TextFieldGroup";
+import { connect } from "react-redux";
+import PropTypes from "prop-types";
+import { addTransaction } from "../../actions/portfolioActions";
+
+class AddTransaction extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      ticker: "",
+      units: "",
+      price: "",
+      date: "",
+      errors: {},
+      disabled: false
+    };
+
+    this.onChange = this.onChange.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.errors) {
+      this.setState({ errors: nextProps.errors });
+    }
+  }
+
+  onSubmit(e) {
+    e.preventDefault();
+
+    const expData = {
+      ticker: this.state.ticker,
+      units: this.state.units,
+      price: this.state.price,
+      date: this.state.date
+    };
+
+    this.props.addTransaction(expData, this.props.history);
+  }
+
+  onChange(e) {
+    this.setState({ [e.target.name]: e.target.value });
+  }
+
+  render() {
+    const { errors } = this.state;
+
+    return (
+      <div className="add-transaction">
+        <div className="container">
+          <div className="row">
+            <div className="col-md-8 m-auto">
+              <Link to="/current-portfolio" className="btn btn-light">
```

Comment:
This navigation will be broken

Reply:
#11 Created this issue to address it later.